invoice/src/main/java/org/killbill/billing/invoice/usage/ContiguousIntervalUsageInArrear.java

```diff
@@ -81,7 +82,8 @@
 
     protected final List<TransitionTime> transitionTimes;
     protected final List<BillingEvent> billingEvents;
-    protected final Map<BillingEvent, Set<String>> allSeenUnitTypes;
+    // Ordering is important here!
+    protected final LinkedHashMap<BillingEvent, Set<String>> allSeenUnitTypes;
```

Comment:
Do we really need this ordering? It seems all you care about is the ability to find the last billing event in the method `addAllSeenUnitTypesFromPrevBillingEvent` ? Shouldn't such last billing event be retrievable from `billingEvents` ?

Reply:
Yes, we could look it up that way. But in that case, the ordering here needs to be swapped: https://github.com/killbill/killbill/pull/1279/files#diff-050672ee3314d0f9a29f5065b853673bR200 (call `addAllSeenUnitTypesFromPrevBillingEvent` before `addBillingEvent`). This felt a bit more sneaky...